### ISSUE TITLE
mta relay ignores specified port number

### ISSUE THREAD

**Issue by rworkman** [OpenSMTPD/OpenSMTPD#378]
TL;DR: I specify port 587 for relaying but 25 gets used instead.

In /etc/opensmtpd/smtpd.conf, I have this:

table secrets file:/etc/opensmtpd/secrets
accept from local sender root@home.rlworkman.net for any relay \
  via tls+auth://rworkman@mail.rlworkman.net:587 auth <secrets>

/etc/opensmtpd/secrets contains "rworkman:password_redacted"

Here's what shows in output of smtpd -dv

smtp-in: Accepted message bec146dc on session ffffffff93a83516: from=root@home.rlworkman.net, to=robby@rlworkman.net, size=314, ndest=1, proto=ESMTP
debug: mta: received evp:bec146dc85b9ab5f for robby@rlworkman.net
debug: mta: draining [relay:rlworkman.net] refcount=1, ntask=1, nconnector=0, nc
onn=0
debug: mta: querying MX for [relay:rlworkman.net]...
debug: mta: [relay:rlworkman.net] waiting for MX
smtp-in: Closing session ffffffff93a83516
debug: smtp: 0x134aaa0: deleting session: done
debug: MXs for domain rlworkman.net:
        207.223.116.211 preference 10
        207.223.116.214 preference 20
debug: mta: ... got mx (0x134a6c0, rlworkman.net, [relay:rlworkman.net])
debug: mta: draining [relay:rlworkman.net] refcount=1, ntask=1, nconnector=0, nconn=0
debug: mta: querying source for [relay:rlworkman.net]...
debug: mta: ... got source for [relay:rlworkman.net]: []
debug: mta: new [connector:[]->[relay:rlworkman.net],0x10000]
debug: mta: connecting with [connector:[]->[relay:rlworkman.net],0x0]
debug: mta-routing: searching new route for [connector:]-> relay:rlworkman.net],0x0]...
debug: mta-routing: selecting candidate route [] <-> 207.223.116.211
debug: mta-routing: spawning new connection on [] <-> 207.223.116.211
debug: mta: 0x1346cd0: spawned for relay [relay:rlworkman.net]
debug: mta: connecting with [connector:[]->[relay:rlworkman.net],0x0]
debug: mta: cannot use [relay:rlworkman.net] before 2s
debug: mta-routing: no route available for [connector:]->[relay:rlworkman.net],0x0]: must wait a bit
debug: mta: retrying to connect on [connector:[]->[relay:rlworkman.net],0x0] in 2s...
debug: mta: draining [relay:rlworkman.net] refcount=3, ntask=1, nconnector=1, nconn=1
debug: mta: scheduling relay [relay:rlworkman.net] in 1s...
[warn] event_del: event has no event_base set.
smtp-out: Connecting to smtp+tls://207.223.116.211:25 (harrier.slackbuilds.org) on session 462bf78a34239426...
[warn] event_del: event has no event_base set.
debug: mta: ... timeout for [relay:rlworkman.net]
...

Looks like the port (if specified) is gotten here (in to.c):
    if ((sep = strrchr(p, ':')) != NULL) {
        *sep = 0;
        relay->port = strtonum(sep+1, 1, 0xffff, &errstr);
Is that second line intentional? I'm not a C guy, but it seems wrong - maybe leftover from testing??. It doesn't make a difference if it's removed though - the problem still exists...
